pkg/parquetquery/predicate_test.go

```diff
@@ -32,41 +33,153 @@ func (p *mockPredicate) KeepValue(parquet.Value) bool             { p.valCalled
 func (p *mockPredicate) KeepPage(parquet.Page) bool               { p.pageCalled = true; return p.ret }
 func (p *mockPredicate) KeepColumnChunk(parquet.ColumnChunk) bool { p.chunkCalled = true; return p.ret }
 
+type predicateTestCase struct {
+	testName   string
+	writeData  func(w *parquet.Writer) //nolint:all
+	keptChunks int
+	keptPages  int
+	keptValues int
+	predicate  Predicate
+}
+
 func TestSubstringPredicate(t *testing.T) {
+	testCases := []predicateTestCase{
+		{
+			testName:   "all chunks/pages/values inspected",
+			predicate:  NewSubstringPredicate("b"),
+			keptChunks: 1,
+			keptPages:  1,
+			keptValues: 2,
+			writeData: func(w *parquet.Writer) { //nolint:all
+				type String struct {
+					S string `parquet:",dict"`
+				}
+				require.NoError(t, w.Write(&String{"abc"})) // kept
+				require.NoError(t, w.Write(&String{"bcd"})) // kept
+				require.NoError(t, w.Write(&String{"cde"})) // skipped
+			},
+		},
+		{
+			testName:   "dictionary in the page header allows for skipping a page",
+			predicate:  NewSubstringPredicate("x"), // Not present in any values
+			keptChunks: 1,
+			keptPages:  0,
+			keptValues: 0,
+			writeData: func(w *parquet.Writer) { //nolint:all
+				type dictString struct {
+					S string `parquet:",dict"`
+				}
+				require.NoError(t, w.Write(&dictString{"abc"}))
+				require.NoError(t, w.Write(&dictString{"abc"}))
+				require.NoError(t, w.Write(&dictString{"abc"}))
+				require.NoError(t, w.Write(&dictString{"abc"}))
+				require.NoError(t, w.Write(&dictString{"abc"}))
+			},
+		},
+	}
+
+	for _, tC := range testCases {
+		t.Run(tC.testName, func(t *testing.T) {
+			testPredicate(t, tC)
+		})
+	}
+}
+
+func TestNewRegexInPredicate(t *testing.T) {
+	testCases := []predicateTestCase{
+		{
+			testName: "all chunks/pages/values inspected",
+			predicate: func() Predicate {
+				pred, err := NewRegexInPredicate([]string{"a.*"})
+				assert.NoError(t, err)
```

Comment:
Can you switch these to `require`?  We favor that over `assert` for fail fast and early.